tests/e2e/spec/tonedear.e2e.spec.js

```diff
@@ -0,0 +1,230 @@
+                    }
+                } catch (e) {
+                    // Do nothing
+                }
+
+                if (mode === 'serviceworker') {
+                    // Disable source verification in SW mode as the dialogue box gave inconsistent test results
+                    const sourceVerificationCheckbox = await driver.findElement(By.id('enableSourceVerification'));
+                    if (sourceVerificationCheckbox.isSelected()) {
+                        await sourceVerificationCheckbox.click();
+                    }
+                }
+            });
+
+            it('Load Tonedear archive and verify content', async function () {
+                if (!serviceWorkerAPI) {
+                    console.log('\x1b[33m%s\x1b[0m', '    - Following test skipped:');
+                    this.skip();
+                }
+
+                const archiveFiles = await driver.findElement(By.id('archiveFiles'));
+                await driver.executeScript('arguments[0].style.display = "block";', archiveFiles);
+
+                if (!BROWSERSTACK) {
+                    await archiveFiles.sendKeys(tonedearBaseFile);
+                    await driver.executeScript('window.setLocalArchiveFromFileSelect();');
+                } else {
+                    await driver.executeScript(
+                        'window.setRemoteArchives.apply(this, [arguments[0]]);',
+                        [tonedearBaseFile]
+                    );
+                    await driver.sleep(1300);
+                }
+            });
+
+            it('Should navigate from main page to Android & iOS section', async function () {
+                // Switch to the iframe if the content is inside 'articleContent'
+                await driver.switchTo().frame('articleContent');
+                // console.log('Switched to iframe successfully');
+
+                // Wait until the link "Android & iOS App" is present in the DOM
+                await driver.wait(async function () {
+                    const contentAvailable = await driver.executeScript('return document.querySelector(\'a[href="android-ios-ear-training-app"]\') !== null;');
+                    return contentAvailable;
+                }, 10000); // Increased to 10 seconds for more loading time
+
+                // Find the "Android & iOS App" link
+                const androidLink = await driver.findElement(By.css('a[href="android-ios-ear-training-app"]'));
+
+                // Verify that the element is found
+                // console.log('Android & iOS App link found:', androidLink !== null);
+
+                // Scroll the element into view and click it
+                await driver.executeScript('arguments[0].scrollIntoView(true);', androidLink);
+                await driver.wait(until.elementIsVisible(androidLink), 10000); // Wait until it's visible
+                await androidLink.click();
+
+                // Take a screenshot after clicking for debugging
```

Comment:
Problem with this is that it's impossible to see screenshots in GitHub Actions AFAIK!